core/arch/x86/emit_utils.c

```diff
@@ -1354,13 +1354,59 @@ append_restore_simd_reg(dcontext_t *dcontext, instrlist_t *ilist, bool absolute)
         int i;
         uint opcode = move_mm_reg_opcode(true /*align16*/, true /*align32*/);
         ASSERT(proc_has_feature(FEATURE_SSE));
-        for (i = 0; i < proc_num_simd_saved(); i++) {
+        instr_t *post_restore = NULL;
```

Comment:
How about `if (!preserve_xmm_caller_saved()) return` to avoid all this indenting.

Reply:
I would do that even more often, even for smaller if-blocks. Done.